### PR TITLE
DB Service (clients)

### LINKED ISSUE #21: Client DAO

### PR DESCRIPTION
**Pull request by maclewan**
Closes #21

### CODE REVIEW

File: app/src/main/java/com/piwniczna/mojakancelaria/DB/DAO.kt [hunk 1]
```diff
@@ -20,19 +18,83 @@ interface DAO {
...
+    @Query("DELETE FROM payments")
+    fun clearPayments()
+
+    @Query("DELETE FROM relations")
+    fun clearRelations()
+
+
+    //clear by id
+    @Query("DELETE FROM clients WHERE id = :clientId")
+    fun clearUser(clientId: Int)
```
Comment (wiktoriasalamon) [inline app/src/main/java/com/piwniczna/mojakancelaria/DB/DAO.kt]: client!!

File: app/src/main/java/com/piwniczna/mojakancelaria/Models/RelationEntity.kt [hunk 2]
```diff
@@ -2,16 +2,22 @@ package com.piwniczna.mojakancelaria.Models
 
 import androidx.room.ColumnInfo
 import androidx.room.Entity
+import androidx.room.ForeignKey
 import androidx.room.PrimaryKey
 import java.math.BigDecimal
 
 
-@Entity(tableName = "relations")
+@Entity(tableName = "relations", foreignKeys = [ForeignKey(entity = ClientEntity::class,
```
Comment (wiktoriasalamon) [inline app/src/main/java/com/piwniczna/mojakancelaria/Models/RelationEntity.kt]: obligations and payments id's are also foreign keys

File: app/src/main/java/com/piwniczna/mojakancelaria/activities/LoginActivity.kt [hunk 3]
```diff
@@ -1,33 +1,28 @@
...
-import java.lang.Exception
 import java.lang.NullPointerException
 import java.nio.charset.StandardCharsets
 
 
 class LoginActivity : AppCompatActivity() {
     lateinit var passwordEditText : EditText
     lateinit var loginButton: Button
-    lateinit var database : MyDb
+    lateinit var service : DataService
```
Comment (wiktoriasalamon) [inline app/src/main/java/com/piwniczna/mojakancelaria/activities/LoginActivity.kt]: name it as dbService

Comment (maclewan) [reply app/src/main/java/com/piwniczna/mojakancelaria/activities/LoginActivity.kt]: fixed

File: app/src/main/java/com/piwniczna/mojakancelaria/Models/RelationEntity.kt (hunk 2 above)
Comment (maclewan) [reply app/src/main/java/com/piwniczna/mojakancelaria/Models/RelationEntity.kt]: wixed

File: app/src/main/java/com/piwniczna/mojakancelaria/DB/DAO.kt (hunk 1 above)
Comment (maclewan) [reply app/src/main/java/com/piwniczna/mojakancelaria/DB/DAO.kt]: fixed

**Pull request merged**